# Written and placed in public domain by Jeffrey Walton.
# This patch fixes some issues with Nettle.
--- run-tests
+++ run-tests
@@ -44,6 +44,13 @@
   export WINEPATH
 fi
 
+nettle_libdir=`dirname $PWD`/.lib
+LD_LIBRARY_PATH=$nettle_libdir
+DYLD_LIBRARY_PATH=$nettle_libdir
+
+export LD_LIBRARY_PATH
+export DYLD_LIBRARY_PATH
+
 # When used in make rules, we sometimes get the filenames VPATH
 # expanded, but usually not.
 find_program () {

--- testsuite/nettle-pbkdf2-test
+++ testsuite/nettle-pbkdf2-test
@@ -4,6 +4,13 @@
   srcdir=`pwd`
 fi
 
+nettle_libdir=`dirname $PWD`/.lib
+LD_LIBRARY_PATH=$nettle_libdir
+DYLD_LIBRARY_PATH=$nettle_libdir
+
+export LD_LIBRARY_PATH
+export DYLD_LIBRARY_PATH
+
 test_pbkdf2 () {
     password="$1"
     salt="$2"
--- testsuite/sexp-conv-test
+++ testsuite/sexp-conv-test
@@ -4,6 +4,13 @@
   srcdir=`pwd`
 fi
 
+nettle_libdir=`dirname $PWD`/.lib
+LD_LIBRARY_PATH=$nettle_libdir
+DYLD_LIBRARY_PATH=$nettle_libdir
+
+export LD_LIBRARY_PATH
+export DYLD_LIBRARY_PATH
+
 print_raw () {
     printf "%s" "$1" > "$2"
 }
--- testsuite/pkcs1-conv-test
+++ testsuite/pkcs1-conv-test
@@ -4,6 +4,13 @@
   srcdir=`pwd`
 fi
 
+nettle_libdir=`dirname $PWD`/.lib
+LD_LIBRARY_PATH=$nettle_libdir
+DYLD_LIBRARY_PATH=$nettle_libdir
+
+export LD_LIBRARY_PATH
+export DYLD_LIBRARY_PATH
+
 [ -x ../tools/pkcs1-conv$EXEEXT ] || exit 77
 
 # Private RSA key, generated by openssl
--- examples/setup-env
+++ examples/setup-env
@@ -2,6 +2,13 @@
 
 set -e
 
+nettle_libdir=`dirname $PWD`/.lib
+LD_LIBRARY_PATH=$nettle_libdir
+DYLD_LIBRARY_PATH=$nettle_libdir
+
+export LD_LIBRARY_PATH
+export DYLD_LIBRARY_PATH
+
 if [ -x rsa-keygen$EXEEXT ] ; then
   $EMULATOR ./rsa-keygen -r rsa-decrypt$EXEEXT -o testkey || exit 1
 fi
--- examples/rsa-encrypt-test
+++ examples/rsa-encrypt-test
@@ -4,6 +4,13 @@
   srcdir=`pwd`
 fi
 
+nettle_libdir=`dirname $PWD`/.lib
+LD_LIBRARY_PATH=$nettle_libdir
+DYLD_LIBRARY_PATH=$nettle_libdir
+
+export LD_LIBRARY_PATH
+export DYLD_LIBRARY_PATH
+
 data="$srcdir/nettle-benchmark.c"
 
 if [ -x rsa-encrypt$EXEEXT ] ; then
--- examples/rsa-verify-test
+++ examples/rsa-verify-test
@@ -4,6 +4,13 @@
   srcdir=`pwd`
 fi
 
+nettle_libdir=`dirname $PWD`/.lib
+LD_LIBRARY_PATH=$nettle_libdir
+DYLD_LIBRARY_PATH=$nettle_libdir
+
+export LD_LIBRARY_PATH
+export DYLD_LIBRARY_PATH
+
 data="$srcdir/nettle-benchmark.c"
 
 if [ -x rsa-verify$EXEEXT ] ; then
--- examples/rsa-sign-test
+++ examples/rsa-sign-test
@@ -4,6 +4,13 @@
   srcdir=`pwd`
 fi
 
+nettle_libdir=`dirname $PWD`/.lib
+LD_LIBRARY_PATH=$nettle_libdir
+DYLD_LIBRARY_PATH=$nettle_libdir
+
+export LD_LIBRARY_PATH
+export DYLD_LIBRARY_PATH
+
 data="$srcdir/nettle-benchmark.c"
 
 if [ -x rsa-sign$EXEEXT ] ; then
--- examples/nettle-openssl.c
+++ examples/nettle-openssl.c
@@ -374,6 +374,11 @@
   EVP_DigestUpdate(ctx->evp, src, length);
 }
 
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
+# define EVP_MD_CTX_new   EVP_MD_CTX_create
+# define EVP_MD_CTX_free  EVP_MD_CTX_destroy
+#endif
+
 #define OPENSSL_HASH(NAME, name)					\
 static void								\
 openssl_##name##_init(void *p)						\
